core/templates/pages/topics-and-skills-dashboard-page/topics-list/topics-list.component.ts

```diff
@@ -115,11 +115,36 @@ export class TopicsListComponent {
     return topicSerialNumber + 1;
   }
 
+  /**
+   ** @param {Number} topicId - ID of the topic.
+   * @returns {CreatorTopicSummary} The topic summary of the topic
+   * of the topic with given topicId.
+   */
+  getTopicSummaryForTopicId(topicId: string): CreatorTopicSummary | undefined {
+    const topicSummary = this.topicSummaries.find(
+      (topicSummary: CreatorTopicSummary) => {
+        return topicSummary.id === topicId;
+      }
+    );
+
+    return topicSummary;
+  }
+
   /**
    * @param {String} topicId - ID of the topic.
    * @param {String} topicName - Name of the topic.
    */
   deleteTopic(topicId: string, topicName: string): void {
+    const topicSummary = this.getTopicSummaryForTopicId(topicId);
+
+    if (topicSummary && topicSummary.classroom) {
```

Comment:
if (topicSummary?.classroom) should work?

Reply:
Done.